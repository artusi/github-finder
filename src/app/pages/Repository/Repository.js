import React from "react";
import PropTypes from "prop-types";
import Header from "components/Header";
import RepoInfo from "components/RepoInfo";
import CommitsList from "components/CommitsList";
import { SubmitButton } from "ui";
import styles from "./Repository.module.css";

export function Repository(props) {
  const { params } = props.match;

  return (
    <div className={styles.repository}>
      <Header />
      <div className={styles.info}>
        <RepoInfo
          username={params.username}
          repositoryName={params.repository}
        />
        <SubmitButton
          id="btnReturnToListRepo"
          onClick={() => {
            props.history.push(`/${params.username}`);
          }}
        >
          Voltar
        </SubmitButton>
      </div>
      <div className={styles.report}>
        <CommitsList />
      </div>
    </div>
  );
}

Repository.propTypes = {
  match: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};

export default Repository;
