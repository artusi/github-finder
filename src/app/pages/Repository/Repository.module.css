.repository {
}
.report {
  margin: 16px 0;
  text-align: left;
}
.info {
  display: flex;
  justify-content: space-between;
}

@media (max-width: 400px) {
  .report {
    margin: 16px 8px;
  }
  .info {
    margin: 0 8px;
  }
}
