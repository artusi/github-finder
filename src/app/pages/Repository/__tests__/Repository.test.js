import React from "react";
import { Repository } from "../Repository";

describe("<Repository />", () => {
  it("renders", () => {
    const wrapper = shallow(
      <Repository
        history={{}}
        match={{ params: { username: "artusi", repository: "git-finder" } }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
