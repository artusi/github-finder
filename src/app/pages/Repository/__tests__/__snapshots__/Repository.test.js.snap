// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository /> renders 1`] = `
<div
  className="repository"
>
  <Header />
  <div
    className="info"
  >
    <Connect(RepoInfo)
      repositoryName="git-finder"
      username="artusi"
    />
    <SubmitButton
      customClassName=""
      disabled={false}
      id="btnReturnToListRepo"
      onClick={[Function]}
    >
      Voltar
    </SubmitButton>
  </div>
  <div
    className="report"
  >
    <Connect(CommitsList) />
  </div>
</div>
`;
